tests/test_middleware_interface.py

```diff
@@ -188,6 +195,8 @@ def test_ingest_image(self):
             datasets = list(self.butler.registry.queryDatasets('raw',
                                                                collections=[f'{instname}/raw/all']))
             self.assertEqual(datasets[0].dataId, data_id)
+            # TODO: I think we also need to check that the visits are defined?
+            # Or should that happen in prep_butler?
```

Comment:
We _should_ define visits ASAP, but it's impossible to do it before `exposure` dimension records are defined, and at present it's very hard to do that before raws are ingested. See my note in the sprint summary document.